File: src/main/java/com/skylotoneoneone/service/AlertService.java
package com.skylotoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylotoneoneone.domain.Alert;
import com.skylotoneoneone.dto.AlertDTO;
import com.skylotoneoneone.dto.AlertSearchDTO;
import com.skylotoneoneone.dto.AlertPageDTO;
import com.skylotoneoneone.dto.AlertConvertCriteriaDTO;
import com.skylotoneoneone.service.GenericService;
import com.skylotoneoneone.dto.common.RequestDTO;
import com.skylotoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AlertService extends GenericService<Alert, Integer> {

	List<Alert> findAll();

	ResultDTO addAlert(AlertDTO alertDTO, RequestDTO requestDTO);

	ResultDTO updateAlert(AlertDTO alertDTO, RequestDTO requestDTO);

    Page<Alert> getAllAlerts(Pageable pageable);

    Page<Alert> getAllAlerts(Specification<Alert> spec, Pageable pageable);

	ResponseEntity<AlertPageDTO> getAlerts(AlertSearchDTO alertSearchDTO);
	
	List<AlertDTO> convertAlertsToAlertDTOs(List<Alert> alerts, AlertConvertCriteriaDTO convertCriteria);

	AlertDTO getAlertDTOById(Integer alertId);







}

File: skylotoneoneone_ui/src/services/AlertService.js
import http from "../http-common"; 

class AlertService {
  getAllAlerts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/alert/alerts`, searchDTO);
  }

  get(alertId) {
    return this.getRequest(`/alert/${alertId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/alert?field=${matchData}`, null);
  }

  addAlert(data) {
    return http.post("/alert/addAlert", data);
  }

  update(data) {
  	return http.post("/alert/updateAlert", data);
  }
  
  uploadImage(data,alertId) {
  	return http.postForm("/alert/uploadImage/"+alertId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AlertService();
